curvefs/src/client/inode_cache_manager.cpp

```diff
@@ -348,33 +407,53 @@ void InodeCacheManagerImpl::ReleaseCache(uint64_t parentId) {
 void InodeCacheManagerImpl::FlushInodeBackground() {
     LOG(INFO) << "flush thread is start.";
     while (!isStop_.load()) {
-        FlushInodeOnce();
-        sleeper_.wait_for(std::chrono::seconds(flushPeriodSec_));
         if (iCache_->Size() > maxCacheSize_) {
             TrimIcache(iCache_->Size() - maxCacheSize_);
         }
+        FlushInodeOnce();
+        sleeper_.wait_for(std::chrono::seconds(flushPeriodSec_));
     }
     LOG(INFO) << "flush thread is stop.";
 }
 
+void InodeCacheManagerImpl::RemoveICache(
+    const std::shared_ptr<InodeWrapper> &inode) {
+    if (!inode->IsDirty() && inode->S3ChunkInfoEmpty()) {
```

Comment:
Here should get lock before judging IsDirty and S3ChunkInfoEmpty

Reply:
fixed